### PR TITLE
Add variables for task and container CPU and memory

### PR DESCRIPTION
**Pull request by crbudzeak**
Ishmael asked that the CPU and memory parameters for the task and container be overridable variables. His team's use case requires a lot more of both to execute successfully.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -83,4 +83,28 @@ variable "permissions_boundary" {
...
+}
+
+variable "task_cpu" {
+  description = "The ECS Task CPU size"
+  type        = number
+  default     = 256
+}
+
+variable "task_memory" {
+  description = "The ECS Task memroy size"
```
Comment (leslie-corbalt) [inline variables.tf:95]: typo: memory

File: variables.tf [hunk 2]
```diff
@@ -83,4 +83,28 @@ variable "permissions_boundary" {
   description = "ARN of the policy that is used to set the permissions boundary for the role"
   type        = string
   default     = ""
+}
+
+variable "task_cpu" {
+  description = "The ECS Task CPU size"
+  type        = number
+  default     = 256
```
Comment (leslie-corbalt) [inline variables.tf:91]: Should the default task cpu value be `512` if the default task memory is `1024` per the task memory/cpu combinations here: [aws task definition params](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html)?
The default values for task_cpu and task_memory should be a valid combination.

**Review by leslie-corbalt — COMMENTED** [review]
Not related to this change, but I have a question about line 233 in ecs.tf: is the `for` expression doing anything?

I made 2 small comments.

File: variables.tf (hunk 1 above)
Comment (crbudzeak) [reply variables.tf:95]: Oops, thanks!

File: variables.tf (hunk 2 above)
Comment (crbudzeak) [reply variables.tf:91]: I kept the hardcoded values in the terraform before moving them to variables. Though, I double-checked the values for valid combinations and found the same page. According to the page, a CPU value of 256 can have a memory value of 512MB, 1 GB, or 2GB, as shown in the screenshot below. Is there something there that I am misunderstanding?

Screen Shot 2022-01-12 at 8 54 47 AM

**Comment by crbudzeak** [comment]
About the `for` in ecs.tf. I expect that it does set the subnet IDs from the variable list. I haven't verified that.

File: variables.tf (hunk 2 above)
Comment (leslie-corbalt) [reply variables.tf:91]: > I kept the hardcoded values in the terraform before moving them to variables. Though, I double-checked the values for valid combinations and found the same page. According to the page, a CPU value of 256 can have a memory value of 512MB, 1 GB, or 2GB, as shown in the screenshot below. Is there something there that I am misunderstanding?
> 
> Screen Shot 2022-01-12 at 8 54 47 AM

You are right. Memory size of 1024 is 1 GB, so the hardcoded combination of cpu/memory is correct.

File: variables.tf [hunk 3]
```diff
@@ -83,4 +83,28 @@ variable "permissions_boundary" {
...
+}
+
+variable "container_cpu" {
+  description = "The container CPU size"
+  type        = number
+  default     = 128
+}
+
+variable "container_memory" {
+  description = "The container memroy size"
```
Comment (leslie-corbalt) [inline variables.tf]: just saw this typo!

**Review by leslie-corbalt — COMMENTED** [review]
typo shows up in 1 more place

**Review by leslie-corbalt — APPROVED** [review]
You fixed all typos. LGTM!

**Pull request merged**